chain/client/src/sync/external.rs

```diff
@@ -222,20 +251,32 @@ pub fn external_storage_location_directory(
     epoch_id: &EpochId,
     epoch_height: u64,
     shard_id: u64,
+    obj_type: ObjectType,
 ) -> String {
-    location_prefix(chain_id, epoch_height, epoch_id, shard_id)
+    location_prefix(chain_id, epoch_height, epoch_id, shard_id, obj_type)
 }
 
 pub fn location_prefix(
     chain_id: &str,
     epoch_height: u64,
     epoch_id: &EpochId,
     shard_id: u64,
+    obj_type: ObjectType,
 ) -> String {
-    format!(
-        "chain_id={}/epoch_height={}/epoch_id={}/shard_id={}",
-        chain_id, epoch_height, epoch_id.0, shard_id
-    )
+    match obj_type {
+        ObjectType::StatePart => format!(
+            "chain_id={}/epoch_height={}/epoch_id={}/shard_id={}",
+            chain_id, epoch_height, epoch_id.0, shard_id
+        ),
+        ObjectType::StateHeader => format!(
+            "chain_id={}/epoch_height={}/epoch_id={}/headers/shard_id={}",
```

Comment:
I think it's ok for now but generally speaking such migrations should be made backwards compatible. During release the nodes in the network will be running two different versions of neard and so changes in path will likely breaks for the nodes that are on different version than the dumping node.